coffeecard/CoffeeCard/Helpers/MobilePay/MobilePayException.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Net;
+using CoffeeCard.Helpers.MobilePay.ErrorMessage;
+
+namespace CoffeeCard.Helpers.MobilePay
+{
+    [Serializable]
+    public class MobilePayException : Exception
+    {
+        private readonly IMobilePayErrorMessage _errorMessage;
```

Comment:
It is not possible to use the existing Message field in exceptions in some manner, to avoid confusion when working with the exception?

Reply:
The super class takes a string in it's constructor where we already store the error message. In the `GetErrorMessage()` we return the same error message which we already set in the super class, so I guess youre right there is no need to have it as a field :)